auth/standard_role_manager.cc

```diff
@@ -266,9 +270,14 @@ future<> standard_role_manager::start() {
                             return;
                         }
                     }
-                    create_default_role_if_missing().get();
+                    create_default_role_if_missing().then([legacy, work_done] mutable {
+                        if (!legacy) {
```

Comment:
please avoid branching on legacy, it's written currently so that it'd be easy to drop it in the future, it's better to leave old flow intact.

currently the structure is usually:

if (legacy_mode) {
// old path
return
}
// new path